src/box/lua/net_box.lua

```diff
@@ -772,7 +800,19 @@ end
 
 function remote_methods:ping(opts)
     check_remote_arg(self, 'ping')
-    return (pcall(self._request, self, M_PING, opts, nil, self._stream_id))
+    local status, ret, err = pcall(self._request_ex, self, M_PING,
+                                   opts, nil, self._stream_id)
+    if not status and ret then
+        -- error occurred in remote_methods:_request due to incorrect
+        -- arguments or fiber. It's not related to underneath transport.
+        error(ret)
+    end
+    -- return only one value in case of no error.
+    -- Mirrror the behavior of future's return
+    if not ret then
+        return ret, err
```

Comment:
Raising an error on invalid arguments is okay, but changing the return value looks like a breaking change - `ping` used to return `true` or `false` while now it returns `true` or `nil, error`. Do we really want this change?